### PR TITLE
rename to bmc-catcher, use go-hal and send latest BMC information on report

### PR DESCRIPTION
**Pull request by kolsa**
Renaming to bmc-catcher may be applied as well.
WDYT?

### CODE REVIEW

File: internal/ipmi/uuid.go [hunk 1]
```diff
@@ -90,26 +90,18 @@ func parseUUIDLine(l string) string {
...
-		err = cmd.Wait()
-		if err != nil {
-			u.log.Errorw("loadUUID", "ip", ip, "wait error", err)
-		}
-	}()
-	s := bufio.NewScanner(out)
-	for s.Scan() {
-		l := s.Text()
+
+	for _, l := range info {
```
Comment (majst01) [inline internal/ipmi/uuid.go:104]: I thought these informations where also available via the redfish api, and then we do not need to make a supermicro proprietary sum call to get them.

Comment (kolsa) [reply internal/ipmi/uuid.go:104]: Then we need the redfish client also for supermicro. I'll check this out

Comment (kolsa) [reply internal/ipmi/uuid.go:104]: I'll also check if we can get the machine UUID for supermicro via redfish as well.
If both works, go-hal does not need to use the sum tool anymore.

**Comment by kolsa** [comment]
merge as soon as go-hal is tagged and used here

**Review by majst01 — APPROVED** [review]

**Review by Gerrit91 — APPROVED** [review]

**Comment by majst01** [comment]
One question arises, i  can remember that for some boards the uuid gathered with redfish comes in BigEndian instead of LittleEndian (or vice versa). @kolsa  can you please check if the uuid is the same with this approach, at least for our 2 different machine types.

File: go.mod [hunk 2]
```diff
@@ -1,17 +1,14 @@
...
+module github.com/metal-stack/bmc-catcher
 
 go 1.14
 
 require (
-	github.com/go-openapi/spec v0.19.8 // indirect
 	github.com/kelseyhightower/envconfig v1.4.0
-	github.com/metal-stack/metal-go v0.7.6
+	github.com/metal-stack/go-hal v0.1.12-0.20201001083047-90e1043cb849
+	github.com/metal-stack/metal-go v0.9.3-0.20201012065727-c724c2699a32
```
Comment (Gerrit91) [inline go.mod:8]: This has to be pinned to a fixed version before merging.

Here are the relevant PRs:
- https://github.com/metal-stack/metal-api/pull/114
- https://github.com/metal-stack/go-hal/pull/10

File: Dockerfile [hunk 3]
```diff
@@ -1,12 +1,8 @@
 FROM metalstack/builder:latest as builder
 
-FROM registry.fi-ts.io/metal/supermicro:2.4.0 as sum
-FROM debian:10
```
Comment (majst01) [inline Dockerfile:4]: by accident ? Should still derive from debian:10

File: go.mod [hunk 4]
```diff
@@ -1,17 +1,14 @@
...
+module github.com/metal-stack/bmc-catcher
 
 go 1.14
 
 require (
-	github.com/go-openapi/spec v0.19.8 // indirect
 	github.com/kelseyhightower/envconfig v1.4.0
-	github.com/metal-stack/metal-go v0.7.6
+	github.com/metal-stack/go-hal v0.2.1
+	github.com/metal-stack/metal-go v0.9.3-0.20201028091601-1ac67a21e174
```
Comment (majst01) [inline go.mod:8]: needs to be updated to a fixed version

**Pull request merged**